File: Java/AutomationFramework/src/com/app/helpers/actions/WindowActions.java
package com.app.helpers.actions;

import org.openqa.selenium.WebDriver;

public class WindowActions {
    WebDriver driver;

    WindowActions(WebDriver driver)
    {
        this.driver = driver;
    }

    public String CurrentWindow()
    {
        return driver.getWindowHandle();
    }

    public void SwitchWindow()
    {
        for(String win: driver.getWindowHandles())
        {
            driver.switchTo().window(win);
        }
    }

    public void SwitchWindowByName(String window)
    {
        driver.switchTo().window(window);
    }
}

File: DotNet/Framework/AutomationFramework/Helpers/Actions/WindowActions.cs
using OpenQA.Selenium;

namespace Helpers.Actions
{
    public class WindowActions
    {
        IWebDriver driver;

        public WindowActions(IWebDriver driver)
        {
            this.driver = driver;
        }

        public string CurrentWindow()
        {
            return driver.CurrentWindowHandle;
        }

        public void SwitchWindow()
        {
            foreach (var window in driver.WindowHandles)
            {
                driver.SwitchTo().Window(window);
            }
        }

        public void SwitchWindowByName(string window)
        {
            driver.SwitchTo().Window(window);
        }
    }
}
